src/pages/announcements/CreateAnnouncementDialog.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FunctionComponent, useCallback } from 'react';
+import {
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    Grid,
+} from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import { FormProvider, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { DurationInput } from './DurationInput';
+import { SubmitButton, TextInput } from '@gridsuite/commons-ui';
+import {
+    AnnouncementFormData,
+    emptyFormData,
+    formSchema,
+    fromFrontToBack,
+    MESSAGE,
+} from './utils';
+import { UserAdminSrv } from '../../services';
+
+interface CreateAnnouncementDialogProps {
+    open: boolean;
+    onClose: () => void;
+}
+
+export const CreateAnnouncementDialog: FunctionComponent<
+    CreateAnnouncementDialogProps
+> = (props) => {
+    const formMethods = useForm<AnnouncementFormData>({
+        defaultValues: emptyFormData,
+        //@ts-ignore because yup TS is broken
+        resolver: yupResolver(formSchema),
+    });
+
+    const { handleSubmit, reset } = formMethods;
+
+    const onSubmit = useCallback(
+        (formData: AnnouncementFormData) => {
+            UserAdminSrv.createAnnouncement(fromFrontToBack(formData));
+            reset();
+            props.onClose();
+        },
+        [props, reset]
+    );
+
+    return (
+        //@ts-ignore because RHF TS is broken
+        <FormProvider validationSchema={formSchema} {...formMethods}>
+            <Dialog open={props.open} onClose={props.onClose}>
```

Comment:
you can unwrap {open, onClose} in the method argument directly

Reply:
I prefer like this then we know it comes from the props. Maybe something to define in the team I agree.